cmake/ConfigureCompiler.cmake

```diff
@@ -354,10 +354,15 @@ else()
       -fno-builtin-free)
   endif()
 
-  if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64" OR CMAKE_SYSTEM_PROCESSOR MATCHES "arm64")
     # Graviton or later
     # https://github.com/aws/aws-graviton-gettting-started
     add_compile_options(-march=armv8-a+crc+simd)
+    if(APPLE)
+      # Boost coroutine currently doesn't support Apple Silicon, so we have to fall
+      # back to the slower ucontext implementation
+      add_compile_definitions(BOOST_USE_UCONTEXT)
```

Comment:
One thing I am not quite clear is that, previously we just use this code to download and deploy a local boost library:

```
find_package(Boost 1.72)

if(Boost_FOUND)
  add_library(boost_target INTERFACE)
  target_link_libraries(boost_target INTERFACE Boost::boost)
else()
  include(ExternalProject)
  ExternalProject_add(boostProject
    URL "https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2"
    URL_HASH SHA256=59c9b274bc451cf91a9ba1dd2c7fdcaf5d60b1b3aa83f2c9fa143417cc660722
    CONFIGURE_COMMAND ""
    BUILD_COMMAND ""
    BUILD_IN_SOURCE ON
    INSTALL_COMMAND ""
    UPDATE_COMMAND ""
    BUILD_BYPRODUCTS <SOURCE_DIR>/boost/config.hpp)

  ExternalProject_Get_property(boostProject SOURCE_DIR)

  set(BOOST_INCLUDE_DIR ${SOURCE_DIR})
  message(STATUS "Boost include dir ${BOOST_INCLUDE_DIR}")

  add_library(boost_target INTERFACE)
  add_dependencies(boost_target boostProject)
  target_include_directories(boost_target SYSTEM INTERFACE ${BOOST_INCLUDE_DIR})
endif()
```

Can we switch to this strategy rather than just injecting boost code into our code?

Reply:
The problem here is that boost coroutine2 depends on boost context - which is a library that needs to be compiled (so far we have only used header-only libraries). I agree that building this through `ExternalProject_Add` is preferable, but this will need some more effort (I might still do this - but I am working on this in my free time so this is more a "I'll do it when I feel like it" kind of situation ;) )

Reply:
to make this even worse, it doesn't seem that boost compiles on my machine if I use the boost build system... Overall it might be easier to just wait for other stuff to support Apple Silicon (before I continue down the path of patching all of our dependencies...)

Reply:
I gotcha... I agree BJam is indeed hard to use -- I always use Debian prebuilt binaries.